### PR TITLE
Feature/GetCode

### PR DESCRIPTION
**Pull request by AnnRiera**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/blockcoders/near-rpc-providers/blob/main/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/near-rpc-provider.spec.ts [hunk 1]
```diff
@@ -71,6 +71,13 @@ describe('NearRpcProvider', () => {
   //   })
   // })
 
+  describe('getCode', () => {
+    it('should get the contract code', async () => {
```
Comment (jarcodallo) [inline src/near-rpc-provider.spec.ts:75]: Could you please add test for get the code by block id and block hash too?

File: src/near-rpc-provider.ts [hunk 2]
```diff
@@ -231,4 +232,30 @@ export class NearRpcProvider extends JsonRpcProvider {
...
+      request_type: 'view_code',
+      account_id: addressOrName,
+    }
+
+    if (blockTag === 'latest') {
+      getCodeParams.finality = 'final'
+    } else {
+      getCodeParams.block_id = blockTag
+    }
+
```
[the remark this answers is not in the document]
Comment (AnnRiera) [reply src/near-rpc-provider.ts:247]: Oh, got it!

File: src/near-rpc-provider.spec.ts (hunk 1 above)
Comment (AnnRiera) [reply src/near-rpc-provider.spec.ts:87]: On it!

File: src/near-rpc-provider.spec.ts [hunk 3]
```diff
@@ -71,6 +71,25 @@ describe('NearRpcProvider', () => {
...
+  describe('getCode', () => {
+    it('should get the contract block by id', async () => {
+      const status = await provider.status()
+      const block = await provider.getCode('blockcoders.testnet', status.sync_info.latest_block_height)
+      expect(block).to.not.be.undefined
+    })
+
+    it('should get the contract block by hash', async () => {
+      const status = await provider.status()
+      const block = await provider.getCode('blockcoders.testnet', status.sync_info.latest_block_hash)
```
Comment (jarcodallo) [inline src/near-rpc-provider.spec.ts:83]: Do you mean code? 👀

File: src/near-rpc-provider.spec.ts [hunk 4]
```diff
@@ -71,6 +71,25 @@ describe('NearRpcProvider', () => {
   //   })
   // })
 
+  describe('getCode', () => {
+    it('should get the contract block by id', async () => {
+      const status = await provider.status()
+      const block = await provider.getCode('blockcoders.testnet', status.sync_info.latest_block_height)
```
Comment (jarcodallo) [inline src/near-rpc-provider.spec.ts:77]: Do you mean code?

**Review by jarcodallo — APPROVED** [review]

**Review by jarcodallo — APPROVED** [review]

**Pull request merged**